lib/Bindings/Python/OMModule.cpp

```diff
@@ -227,6 +257,11 @@ void circt::python::populateDialectOMSubmodule(py::module &m) {
            [](MlirAttribute arr) { return PyListAttrIterator(arr); });
   PyListAttrIterator::bind(m);
 
+  mlir_attribute_subclass(m, "MapAttr", omAttrIsAMapAttr)
+      .def("__iter__",
+           [](MlirAttribute arr) { return PyMapAttrIterator(arr); });
```

Comment:
Could add `__getattr__` and `__len__`?

Reply:
There is not need for `__getattr__` now since this object is not exposed to users as it is converted by `attribute_to_var`. We can consider the method later.